Psedopeople Census ACS Link Example

tylerdy opened the pull request:
PR that doesnt target Splink repo for ReviewNB

Commit:
Initial draft

zmbc commented:
@tylerdy Overall this looks incredible! Really excited about it 🎉 I've left some comments, many of them nit-picky. Let me know if you'd like to talk through in real time.

I'm hoping I can figure out an easy-ish way for me to review this and actually see it rendered as Splink docs, maybe you can help me with that? Have you rendered it locally?

tylerdy commented:
<p>Seems that we do, I dropped these bc I get an error when generating the exploratory analysis charts about missing columns if I don't. I added a comment that thats why I'm doing it</p><p> </p>

---

 <a href='https://app.reviewnb.com/tylerdy/splink/pull/1/discussion/#comment-2762242692'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

tylerdy commented:
<p>Whoops, this was left from FEBRL4 where they had a simple and complex one, I removed it. I think it would be nice to have a more complex model, but not sure its worth doing for ACS.</p>

---

 <a href='https://app.reviewnb.com/tylerdy/splink/pull/1/discussion/#comment-2762242943'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

tylerdy commented:
<p>Hm, from the splink docs I'm not seeing any way to avoid, but yes this is what the Febrl example does.</p>

---

 <a href='https://app.reviewnb.com/tylerdy/splink/pull/1/discussion/#comment-2762243151'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

zmbc commented:
<p>If it's consistent with FEBRL, fine to leave it like this!</p>

---

 <a href='https://app.reviewnb.com/tylerdy/splink/pull/1/discussion/#comment-2762243151'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

tylerdy commented:
<p>Makes sense, got black set up and fixed formatting. Will go through and put in dot syntax where possible</p>

---

 <a href='https://app.reviewnb.com/tylerdy/splink/pull/1/discussion/#comment-2762243252'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

Commit:
Change to using all ACS years, address Zeb's feedback, etc

tylerdy commented:
<p>Hmm I tried rendering docs locally using scripts/make_docs_locally.sh  <a href="https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/contributing_to_docs.html" target="_blank">https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/contributing_to_docs.html</a>, but needed poetry, and it seemed the poetry version required was incompatible with my enviornment, so I tried installing the dependencies with conda <a href="https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/development_quickstart.html#step-3-conda-install-option-install-system-dependencies" target="_blank">https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/development_quickstart.html#step-3-conda-install-option-install-system-dependencies</a> and first had to change line 30 in scripts/conda/development_steup_with_conda.sh to use --yes instead of deprecated --force. Then, I got the error "running bootstrap script ... 2025-04-10 08:28:32.963 PDT [907118] FATAL:&nbsp;data directory "/mnt/share/homes/tylerdy/splink/splink_db" has invalid permissions" which I am guessing is due to running the db on the cluster? </p><p> </p><p><code>+ /ihme/homes/tylerdy/miniforge3/bin/mamba run --no-capture -n splink initdb splink_db</code></p><p><code>The files belonging to this database system will be owned by user "tylerdy".</code></p><p><code>This user must also own the server process.</code></p><p> </p><p><code>The database cluster will be initialized with this locale configuration:</code></p><p><code>&nbsp;locale provider:&nbsp;&nbsp;libc</code></p><p><code>&nbsp;LC_COLLATE:&nbsp;en_US.UTF-8</code></p><p><code>&nbsp;LC_CTYPE:&nbsp;&nbsp;&nbsp;C.UTF-8</code></p><p><code>&nbsp;LC_MESSAGES: en_US.UTF-8</code></p><p><code>&nbsp;LC_MONETARY: en_US.UTF-8</code></p><p><code>&nbsp;LC_NUMERIC:&nbsp;en_US.UTF-8</code></p><p><code>&nbsp;LC_TIME:&nbsp;&nbsp;&nbsp;&nbsp;en_US.UTF-8</code></p><p><code>The default database encoding has accordingly been set to "UTF8".</code></p><p><code>The default text search configuration will be set to "english".</code></p><p> </p><p><code>Data page checksums are disabled.</code></p><p> </p><p><code>creating directory splink_db ... ok</code></p><p><code>creating subdirectories ... ok</code></p><p><code>selecting dynamic shared memory implementation ... posix</code></p><p><code>selecting default "max_connections" ... 25</code></p><p><code>selecting default "shared_buffers" ... 400kB</code></p><p><code>selecting default time zone ... America/Los_Angeles</code></p><p><code>creating configuration files ... ok</code></p><p><code>running bootstrap script ... 2025-04-10 08:28:32.963 PDT [907118] FATAL:&nbsp;data directory "/mnt/share/homes/tylerdy/splink/splink_db" has invalid permissions</code></p><p><code>2025-04-10 08:28:32.963 PDT [907118] DETAIL:&nbsp;Permissions should be u=rwx (0700) or u=rwx,g=rx (0750).</code></p><p><code>child process exited with exit code 1</code></p><p><code>initdb: removing data directory "splink_db"</code></p><p><code>ERROR conda.cli.main_run:execute(125): <code>conda run initdb splink_db</code> failed. (See above for error)</code></p><p> </p><p> </p>

zmbc commented:
Overall great work! In the rendered version I saw today, I also saw a fair amount of horizontal overflow -- can you take a look at subsetting large dataframes to only the most important columns to prevent/limit this?

tylerdy commented:
<p>As far as I can tell, census.year and acs.survey_date need to be called the same thing or I would have to drop both of them. Definitely the exploratory analysis charts failed and I think the model stuff failed as well otherwise. I could call them both "survey" or soemthing vague like that?</p><p> </p>

---

 <a href='https://app.reviewnb.com/tylerdy/splink/pull/1/discussion/#comment-2795577493'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

tylerdy commented:
<p>About the concatenation, I wasn't able to remember this when Nathaniel asked me about this choice in the meeting yesterday, but I think one reason I made the whole address string NaN if any part besides unit_number was NaN was because I had been considering doing more data cleaning where I might be setting invalid values to NaN (eg street numbers that aren't numbers) and I was worried about two addresses incorrectly matching where previously (for example) one had a missing street number and one had a invalid street number like "qwe", or both had different invalid street numbers, etc. </p><p> </p><p>I didn't end up doing any of that cleaning so I dont see any reason not to do it the way you do it in person linkage case study, but I guess wanted to double check my understanding that theres times that it may not be the best way - maybe if its how the census does it, in practice the things I'm talking about don't end up being an issue.</p>

---

 <a href='https://app.reviewnb.com/tylerdy/splink/pull/1/discussion/#comment-2795577501'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

zmbc commented:
<p>Ah sorry, I didn't read this clearly. I think I might prefer dropping them. There just doesn't seem to be a sensible way to make these comparable to one another. </p><p> </p>

---

 <a href='https://app.reviewnb.com/tylerdy/splink/pull/1/discussion/#comment-2795577493'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

tylerdy commented:
<p>Hmm, maybe not, and I had originally dropped them, but now that I'm doing all years for ACS it was nice to have the survey dates in the predictions/data so I could check the ACS respondents that weren't in the 2020 census and show that a lot of them are from the 2037 ACS or another later one and likely moved after the 2020 census was conducted. </p><p> </p><p>But probably the better way to do this would be do add this info to the id lookup table and re insert it into the results/data, just a bit more code to write. Suppose I could also drop the month/day of the ACS column and call them both year, but that feels a little confusing and is still forcing the columns together.</p>

---

 <a href='https://app.reviewnb.com/tylerdy/splink/pull/1/discussion/#comment-2795577493'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

zmbc commented:
<p>I like adding the info to the ID lookup table, or dropping the month/day of the ACS column.</p>

---

 <a href='https://app.reviewnb.com/tylerdy/splink/pull/1/discussion/#comment-2795577493'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

zmbc commented:
<p>That makes sense, and I think you are definitely seeing how any way to do it has pros and cons! Note that the issue you mentioned is true even without extra data cleaning; some e.g. street numbers are missing in the pseudopeople data regardless, and that could cause two addresses that truly differed on that attribute to match when both records are missing that field. These are all kind of hacky workarounds; really, we should have a better approach for comparing this structured address data, but that isn't available to you right now!</p><p> </p>

---

 <a href='https://app.reviewnb.com/tylerdy/splink/pull/1/discussion/#comment-2795577501'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

tylerdy commented:
<p>True, if I were to allow empty parts of an address to still be concatenated into an address string, addresses with missing parts could match - thats a less convoluted way to explain the benefits of doing it the way I did it. Seems like it should be little enough work to change to the person linkage study way to be worth matching for consistency with that.</p>

---

 <a href='https://app.reviewnb.com/tylerdy/splink/pull/1/discussion/#comment-2795577501'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

tylerdy commented:
<p>I decided to leave this for now since I'd have to change how I'm using addresses in some later parts of the model as well, but can come back to it later for consistency with person linkage study if we want.</p>

---

 <a href='https://app.reviewnb.com/tylerdy/splink/pull/1/discussion/#comment-2795577501'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

tylerdy commented:
<p>I limited the max columns for tables and made some of the charts smaller that I could make smaller. Picked some most important columns to show for the predictions tables and put a note about what is hidden.</p>

Commit:
Next round of changes based on feedback

zmbc commented:
@tylerdy All my comments are minor, once you address those we should open the PR on the Splink side! Great work 🎉

tylerdy commented:
<p>Oh true good point about redundancy, but yeah I'll see how it looks with some overflow</p>

---

 <a href='https://app.reviewnb.com/tylerdy/splink/pull/1/discussion/#comment-2803092561'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

Commit:
Respond to next round of feedback and do more thorough proofread

Commit:
Minor change removing best/worst wording about false pos/negs

tylerdy commented:
@zmbc Should be in good shape now, will open PR on Splink side if you want to give one last look for a final sign-off.

zmbc commented:
Let's go over any remaining questions in our meeting, and then open the PR on the Splink side!

I think only the last one is a little tricky. Hadn't thought that through before.

Commit:
Make changes from another round of feedback and ran linter

Commit:
Minor
